pkg/flow/module.go

```diff
@@ -58,6 +59,29 @@ func (m *moduleController) NewModule(id string, export component.ExportFunc) (co
 	return mod, nil
 }
 
+// NewCustomComponent creates a new, unstarted CustomComponent.
+func (m *moduleController) NewCustomComponent(id string, export component.ExportFunc) (controller.CustomComponent, error) {
+	if id != "" && !scanner.IsValidIdentifier(id) {
+		return nil, fmt.Errorf("customComponent ID %q is not a valid River identifier", id)
+	}
+
+	m.mut.Lock()
+	defer m.mut.Unlock()
+	fullPath := m.o.ID
+	if id != "" {
+		fullPath = path.Join(fullPath, id)
+	}
+
+	mod := newModule(&moduleOptions{
+		ID:                      fullPath,
+		export:                  export,
+		moduleControllerOptions: m.o,
+		parent:                  m,
+	})
+
+	return mod, nil
+}
```

Comment:
I don't understand why this method is being added? It looks to be exactly the same as NewModule. Can we avoid adding this, please? If you want to rename NewModule to make it more fitting to both cases, please feel free do so.

Reply:
This new function returns a "controller.CustomComponent" while the other one returns a "component.Module". The idea is that the old one could be removed when we get rid of the old modules. I couldn't reuse the component.Module interface because the LoadOptions struct is defined in the controller package (I could use any but I did not like it). Do you have a suggestion to handle this better?

Reply:
I see. That's an artifact of the registry being passed at apply time instead of at construction time. 

Let's just leave it.